File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Cards/CardGroup.ts
import AbstractCard from "./AbstractCard";

export class CardGroup{
    public Group = new Array<AbstractCard>();
    public Queued = new Array<AbstractCard>();
    public InHand = new Array<AbstractCard>();
    public get length(): number {
        return this.Group.length;
    }
    public forEach(callbackfn: (value: AbstractCard, index: number, array: AbstractCard[]) => void, thisArg?: any): void{
        this.Group.forEach(callbackfn);
    }
    constructor(public Type : CardGroupType){
        
    }
}

export enum CardGroupType {
    DRAW_PILE = "draw_pile",
    MASTER_DECK = "master_deck",
    HAND = "hand",
    DISCARD_PILE = "discard_pile",
    EXHAUST_PILE = "exhaust_pile",
    CARD_POOL = "card_pool",
    UNSPECIFIED = "unspecified"
}

File: Unity/Assets/Scripts/ModTheSpire/Card/CardGroup.cs
using System;
using System.Collections.Generic;

namespace ModTheSpire.Card
{
    public class CardGroup
    {
        public CardGroupType Type;
        public List<AbstractCard> Group = new ();
        public List<AbstractCard> Queued = new();
        public List<AbstractCard> InHand = new();
        public int Length => this.Group.Count;
        public void Foreach(Action<AbstractCard> action){
            this.Group.ForEach(action);
        }

        public CardGroup(CardGroupType type)
        {
            Type = type;
        }

        public void AddToTop(AbstractCard c)
        {
            Group.Add(c);
        }

        public void Discard(bool immedatley = false)
        {
            //飞到对应的位置
        }
    }
}